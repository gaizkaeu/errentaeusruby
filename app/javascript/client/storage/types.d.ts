export interface TaxIncome {
  id: string;
  price: number;
  state: TaxIncomeStatus;
  estimation: string;
  lawyer: string;
  appointment: string;
  created_at: string;
  updated_at: string;
}

export interface Document {
  id: string;
  state: "created" | "uploaded" | "ready";
  name: string;
  description: string;
  tax_income_id: string;
  user_id: string;
  lawyer_id: string;
  created_at: string;
  updated_at: string;
  attachments: DocumentAttachment[];
  export_status: "export_successful" | "export_queue" | "not_exported";
  export: DocumentAttachment;
  document_number: number;
}

export interface DocumentHistory {
  id: string;
  document_id: string;
  user_id: string;
  action: "remove_image" | "add_image" | "completed" | "exported";
  description: string;
  created_at: string;
}

export interface DocumentAttachment {
  filename: string;
  url: string;
  id: string;
}

export interface PaymentDetails {
  status:
    | "succeeded"
    | "canceled"
    | "processing"
    | "requires_action"
    | "requires_confirmation";
  amount: number;
  receipt_url: string;
  card: {
    brand: "visa" | "mastercard";
    last4: string;
    wallet: string | undefined;
    refunded: boolean;
  };
}

export interface Appointment {
  id: string;
  lawyer_id: string;
  tax_income_id: string;
  phone: string;
  time: string;
  method: "office" | "phone";
}

export type TaxIncomeStatus =
  | "pending_assignation"
  | "waiting_for_meeting_creation"
  | "waiting_payment"
  | "waiting_for_meeting"
  | "rejected"
  | "pending_documentation"
  | "in_progress"
  | "finished";

export interface TaxIncomeData {
  observations: string;
  estimation: {
    token: string;
  };
}

export type TaxIncomesResponse = TaxIncome[];

export interface AuthState {
  status: "succeeded" | "loading" | "failed";
  error: string | undefined;
  logged_in: boolean;
  fetched: boolean;
  user: IUser | undefined;
}

export interface IUser {
  id: string;
  email: string;
  first_name: string;
  last_name: string;
  confirmed: boolean;
  account_type: "user" | "lawyer";
}

export interface ValidationErrors {
  error: string;
  errors: Record<string, string>;
}

export interface UserRegistrationData {
  name: string;
  surname: string;
  email: string;
  password: string;
  password_confirmation: string;
}

export interface SessionCreationData {
  email: string;
  password: string;
}

export interface Estimation extends EstimationData {
  price: number;
  id: number;
  first_name: string;
  token: {
    data: string;
    exp: string;
  };
}

export interface EstimationData {
  first_name: string;
  home_changes: number;
  first_time: number;
}

export interface EstimationState {
  status: "succeeded" | "loading" | "failed";
  error: string | undefined;
  estimation_props: EstimationData | undefined;
  estimation: Estimation | undefined;
}

export interface CalculatorValues {
  first_name: string;
  firstTime: string;
  homeChanges: QuestionWithNumber;
  rentalsMortgages: QuestionWithNumber;
  realStateTrade: QuestionWithNumber;
  withCouple: string;
  /*   incomeRent: QuestionWithNumber,
    sharesTrade: QuestionWithNumber, */
  professionalCompanyActivity: string;
}

export interface QuestionWithNumber {
  consta: string;
  numero: number;
}

export interface CalculatorState {
  formValues: CalculatorValues;
  step: number;
}
