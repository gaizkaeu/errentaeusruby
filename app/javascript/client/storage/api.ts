import { BaseQueryFn, createApi } from "@reduxjs/toolkit/query/react";
import axios, { AxiosError, AxiosRequestConfig } from "axios";
import {
  TaxIncomesResponse,
  TaxIncome,
  TaxIncomeData,
} from "./models/TaxIncome";
import {
  Appointment,
  Estimation,
  IUser,
  PaymentDetails,
  Document,
  DocumentHistory,
  UserRegistrationData,
  SessionCreationData,
} from "./types";

const axiosBaseQuery =
  (
    { baseUrl }: { baseUrl: string } = { baseUrl: "" }
  ): BaseQueryFn<
    {
      url: string;
      method: AxiosRequestConfig["method"];
      data?: AxiosRequestConfig["data"];
      params?: AxiosRequestConfig["params"];
    },
    unknown,
    unknown
  > =>
  async ({ url, method, data, params }) => {
    try {
      const result = await axios({ url: baseUrl + url, method, data, params });
      return { data: result.data };
    } catch (axiosError) {
      const err = axiosError as AxiosError;
      return {
        error: { status: err.response?.status, data: err.response?.data },
      };
    }
  };

export const api = createApi({
  reducerPath: "taxIncomeApi",
  baseQuery: axiosBaseQuery({ baseUrl: "/api/v1/" }),
  tagTypes: [
    "TaxIncome",
    "Appointment",
    "Estimation",
    "Lawyer",
    "Document",
    "DocumentHistory",
    "User",
  ],
  endpoints: (build) => ({
    getTaxIncomes: build.query<TaxIncomesResponse, { name: string } | void>({
      query: (data) => ({ url: "tax_incomes", method: "get", params: data }),
      providesTags: (result) =>
        result
          ? [...result.map(({ id }) => ({ type: "TaxIncome" as const, id }))]
          : [{ type: "TaxIncome", id: "LIST" }],
    }),
    getTaxIncomeById: build.query<TaxIncome, string>({
      query: (id) => ({ url: `tax_incomes/${id}`, method: "get" }),
      providesTags: (_result, _error, id) => [{ type: "TaxIncome", id }],
    }),
    createTaxIncome: build.mutation<TaxIncome, TaxIncomeData>({
      query: (data) => ({ url: `tax_incomes`, method: "post", data: data }),
      invalidatesTags: [{ type: "TaxIncome", id: "LIST" }],
    }),
    updateTaxIncome: build.mutation<TaxIncome, Partial<TaxIncome>>({
      query: (data) => ({
        url: `tax_incomes/${data.id}`,
        method: "put",
        data: data,
      }),
      invalidatesTags: (result, _error) => [
        { type: "TaxIncome", id: result?.id },
      ],
    }),
    deleteTaxIncome: build.mutation<void, string>({
      query: (data) => ({
        url: `tax_incomes/${data}`,
        method: "delete",
      }),
      invalidatesTags: (result, _error, id) => [{ type: "TaxIncome", id: id }],
    }),
    getAppointments: build.query<Appointment[], void>({
      query: () => ({ url: "appointments", method: "get" }),
      providesTags: (result) =>
        result
          ? [
              ...result.map(({ id }) => ({ type: "Appointment" as const, id })),
              { type: "Appointment", id: "LIST" },
            ]
          : [{ type: "Appointment", id: "LIST" }],
    }),
    createAppointmentToTaxIncome: build.mutation<
      Appointment,
      Partial<Appointment>
    >({
      query: (data) => ({ url: "appointments", method: "post", data: data }),
      invalidatesTags: (result) => [
        { type: "TaxIncome", id: result?.tax_income_id },
        { type: "Appointment", id: "LIST" },
      ],
    }),
    getAppointmentById: build.query<Appointment, string>({
      query: (id) => ({ url: `appointments/${id}`, method: "get" }),
      providesTags: (_result, _error, id) => [{ type: "Appointment", id }],
    }),
    getEstimationByToken: build.query<Estimation, string>({
      query: (id) => ({
        url: `estimations/estimation_from_jwt`,
        method: "post",
        data: { estimation_jwt: id },
      }),
    }),
    getEstimationById: build.query<Estimation, string>({
      query: (id) => ({ url: `estimations/${id}`, method: "get" }),
      providesTags: (_result, _error, id) => [{ type: "Estimation", id }],
    }),
    getLawyerById: build.query<IUser, string>({
      query: (id) => ({ url: `lawyers/${id}`, method: "get" }),
      providesTags: (_result, _error, id) => [{ type: "Lawyer", id }],
    }),
    updateAppointmentById: build.mutation<Appointment, Partial<Appointment>>({
      query: (data) => ({
        url: `appointments/${data.id}`,
        method: "put",
        params: { appointment: data },
      }),
      invalidatesTags: (result, _error) => [
        { type: "Appointment", id: result?.id },
      ],
    }),
    getPaymentDataOfTaxIncome: build.query<PaymentDetails, string>({
      query: (id) => ({ url: `tax_incomes/${id}/payment_data`, method: "get" }),
    }),
    getDocumentsOfTaxIncome: build.query<Document[], string>({
      query: (id) => ({ url: `tax_incomes/${id}/documents`, method: "get" }),
      providesTags: (result, _error, _arg) =>
        result
          ? [
              ...result.map(({ id }) => ({ type: "Document" as const, id })),
              { type: "Document", id: "LIST" },
            ]
          : [{ type: "Document", id: "LIST" }],
    }),
    getDocumentById: build.query<Document, string>({
      query: (id) => ({ url: `documents/${id}`, method: "get" }),
      providesTags: (_result, _error, id) => [{ type: "Document", id }],
    }),
    deleteDocumentById: build.mutation<void, string>({
      query: (data) => ({
        url: `documents/${data}`,
        method: "delete",
      }),
      invalidatesTags: (result, _error, id) => [{ type: "Document", id: id }],
    }),
    deleteDocumentAttachmentById: build.mutation<
      Document,
      { document_id: string; attachment_id: string }
    >({
      query: (data) => ({
        url: `documents/${data.document_id}/delete_document_attachment/${data.attachment_id}`,
        method: "delete",
      }),
      invalidatesTags: (result, _error) => [
        { type: "Document", id: result?.id },
      ],
    }),
    addAttachmentToDocument: build.mutation<
      Document,
      { document_id: string; files: any }
    >({
      query: (data) => ({
        url: `documents/${data.document_id}/add_document_attachment/`,
        method: "post",
        data: data.files,
      }),
      invalidatesTags: (result, _error) => [
        { type: "Document", id: result?.id },
      ],
    }),
    createDocument: build.mutation<Document, Partial<Document>>({
      query: (data) => ({
        url: `documents`,
        method: "post",
        data: data,
      }),
      invalidatesTags: (_error) => [{ type: "Document", id: "LIST" }],
    }),
    exportDocumentById: build.mutation<Document, string>({
      query: (data) => ({
        url: `documents/${data}/export_document`,
        method: "post",
      }),
      invalidatesTags: (result, _error) => [
        { type: "Document", id: result?.id },
      ],
    }),
    getDocumentHistoryById: build.query<DocumentHistory[], string>({
      query: (id) => ({ url: `documents/${id}/history`, method: "get" }),
    }),
    getCurrentAccount: build.query<IUser, void>({
      query: () => ({ url: `logged_in`, method: "get" }),
      providesTags: (_result, _error, _id) => ["User"],
    }),
    createNewAccount: build.mutation<IUser, UserRegistrationData>({
      query: (data) => ({
        url: "users",
        method: "post",
        data: { api_v1_user: data },
      }),
      invalidatesTags: (_result) => ["User"],
    }),
    loginAccount: build.mutation<IUser, SessionCreationData>({
      query: (data) => ({
        url: "users/sign_in",
        method: "post",
        data: { api_v1_user: data },
      }),
      invalidatesTags: (_result) => ["User"],
    }),
    logOut: build.mutation<void, void>({
      query: () => ({ url: `users/sign_out`, method: "delete" }),
      invalidatesTags: (_result, _error, _id) => ["User"],
    }),
    googleOAuthCallBack: build.mutation<void, string>({
      query: (data) => ({
        url: "users/auth/google_oauth2/callback",
        method: "post",
        data: { code: data },
      }),
      invalidatesTags: (_result) => ["User"],
    }),
    googleOAuthOneTapCallBack: build.mutation<IUser, string>({
      query: (data) => ({
        url: "users/auth/google_one_tap/callback",
        method: "post",
        data: { credential: data },
      }),
      invalidatesTags: (_result) => ["User"],
    }),
    getUserById: build.query<IUser, string>({
      query: (data) => ({ url: `accounts/${data}`, method: "get" }),
    }),
    requestResendConfirmation: build.mutation<{ status: string }, string>({
      query: (data) => ({
        url: `accounts/${data}/resend_confirmation`,
        method: "get",
      }),
    }),
  }),
});

export const {
  useGetTaxIncomesQuery,
  useGetTaxIncomeByIdQuery,
  useCreateTaxIncomeMutation,
  useCreateAppointmentToTaxIncomeMutation,
  useGetAppointmentByIdQuery,
  useGetEstimationByIdQuery,
  useGetLawyerByIdQuery,
  useDeleteDocumentByIdMutation,
  useGetAppointmentsQuery,
  useUpdateAppointmentByIdMutation,
  useGetPaymentDataOfTaxIncomeQuery,
  useGetDocumentsOfTaxIncomeQuery,
  useDeleteDocumentAttachmentByIdMutation,
  useAddAttachmentToDocumentMutation,
  useCreateDocumentMutation,
  useGetDocumentByIdQuery,
  useExportDocumentByIdMutation,
  useGetDocumentHistoryByIdQuery,
  useCreateNewAccountMutation,
  useGetCurrentAccountQuery,
  useLoginAccountMutation,
  useLogOutMutation,
  useGoogleOAuthCallBackMutation,
  useGoogleOAuthOneTapCallBackMutation,
  useGetEstimationByTokenQuery,
  useGetUserByIdQuery,
  useUpdateTaxIncomeMutation,
  useDeleteTaxIncomeMutation,
  useRequestResendConfirmationMutation,
} = api;
