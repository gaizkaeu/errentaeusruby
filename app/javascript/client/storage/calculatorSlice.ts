import { createSlice } from "@reduxjs/toolkit";
import { CalculatorState } from "./types";

const initialState: CalculatorState = {
  formValues: {
    first_name: '',
    homeChanges: {
      consta: '',
      numero: 1,
    },
    rentalsMortgages: {
      consta: '',
      numero: 1,
    },
    realStateTrade: {
      consta: '',
      numero: 1,
    },
    firstTime: '',
    withCouple: '',
    professionalCompanyActivity: '',
  },
  step: 0
}

const calculatorSlice = createSlice({
    name: 'calculator',
    initialState: initialState,
    reducers: {
      valuesChanged(state, payload) {
        state.formValues = payload.payload;
      },
      nextStep(state) {
        state.step++;
      },
      prevStep(state) {
        state.step--;
      },
      firstStep(state) {
        state.step = 0;
      }
    }
})

export default calculatorSlice.reducer;
export const { valuesChanged, nextStep, prevStep, firstStep } = calculatorSlice.actions