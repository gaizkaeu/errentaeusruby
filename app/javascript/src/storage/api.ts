import {BaseQueryFn, buildCreateApi, createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react'
import axios, { AxiosError, AxiosRequestConfig } from 'axios'
import { TaxIncomesResponse, Appointment, TaxIncome, TaxIncomeData, Estimation, IUser } from './types'

const axiosBaseQuery =
  (
    { baseUrl }: { baseUrl: string } = { baseUrl: '' }
  ): BaseQueryFn<
    {
      url: string
      method: AxiosRequestConfig['method']
      data?: AxiosRequestConfig['data']
      params?: AxiosRequestConfig['params']
    },
    unknown,
    unknown
  > =>
  async ({ url, method, data, params }) => {
    try {
      const result = await axios({ url: baseUrl + url, method, data, params })
      return { data: result.data }
    } catch (axiosError) {
      let err = axiosError as AxiosError
      return {
        error: {
          status: err.response?.status,
          data: err.response?.data || err.message,
        },
      }
    }
  }

export const taxIncomeApi = createApi({
  reducerPath: 'taxIncomeApi',
  baseQuery: axiosBaseQuery({baseUrl: '/api/v1/'}),
  tagTypes: ['TaxIncome', 'Appointment', 'Estimation', 'Lawyer'],
  endpoints: (build) => ({
    getTaxIncomes: build.query<TaxIncomesResponse, void>({
      query: () => ({url: 'tax_incomes', method: 'get'}),
      providesTags: (result) =>
        result
          ? [
              ...result.map(({ id }) => ({ type: 'TaxIncome' as const, id })),
              { type: 'TaxIncome', id: 'LIST' },
            ]
          : [{ type: 'TaxIncome', id: 'LIST' }],
    }),
    getTaxIncomeById: build.query<TaxIncome, string>({
        query: (id) => ({url: `tax_incomes/${id}`, method: 'get'}),
        providesTags: (result, error, id) => [{type: 'TaxIncome', id}],
      }),
    createTaxIncome: build.mutation<TaxIncome, TaxIncomeData>({
        query: (data) => ({url: `tax_incomes`, method: 'post', data: data}),
        invalidatesTags: [{ type: 'TaxIncome', id: 'LIST' }],
      }),
    getAppointments: build.query<Appointment[], void>({
      query: () => ({url: 'appointments', method: 'get'}),
      providesTags: (result) =>
        result
          ? [
              ...result.map(({ id }) => ({ type: 'Appointment' as const, id })),
              { type: 'Appointment', id: 'LIST' },
            ]
          : [{ type: 'Appointment', id: 'LIST' }],
    }),
    createAppointmentToTaxIncome: build.mutation<Appointment, Partial<Appointment>>({
      query: (data) => ({url: 'appointments', method: 'post', data: data}),
      invalidatesTags: (result) => [{ type: 'TaxIncome', id:  result?.tax_income_id}, {type: 'Appointment', id: 'LIST'}],
    }), 
    getAppointmentById: build.query<Appointment, string>({
      query: (id) => ({url: `appointments/${id}`, method: 'get'}),
      providesTags: (result, error, id) => [{type: 'Appointment', id}],
    }),
    getEstimationById: build.query<Estimation, string>({
      query: (id) => ({url: `estimations/${id}`, method: 'get'}),
      providesTags: (result, error, id) => [{type: 'Estimation', id}],
    }),
    getLawyerById: build.query<IUser, string>({
      query: (id) => ({url: `lawyers/${id}`, method: 'get'}),
      providesTags: (result, error, id) => [{type: 'Lawyer', id}],
    }),
    updateAppointmentById: build.mutation<Appointment, Partial<Appointment>>({
      query: (data) => ({url: `appointments/${data.id}`, method: 'put', body: data}),
      invalidatesTags: (result, error) => [{ type: 'TaxIncome', id:  result?.tax_income_id}, {type: 'Appointment', id: result?.id}],
    }),
  }),
})

export const { useGetTaxIncomesQuery, useGetTaxIncomeByIdQuery, useCreateTaxIncomeMutation, useCreateAppointmentToTaxIncomeMutation,
               useGetAppointmentByIdQuery, useGetEstimationByIdQuery, useGetLawyerByIdQuery, useGetAppointmentsQuery } = taxIncomeApi