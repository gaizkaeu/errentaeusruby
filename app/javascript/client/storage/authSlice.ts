import { createSlice } from "@reduxjs/toolkit";
import { api } from "./api";
import { AuthState } from "./types";

// Define the initial state using that type
const initialState = {
  status: "succeeded",
  error: "",
  logged_in: false,
  user: undefined,
} as AuthState;

const authSlice = createSlice({
  name: "authentication",
  initialState: initialState,
  reducers: {},
  extraReducers(builder) {
    builder
      .addMatcher(
        api.endpoints.loginAccount.matchFulfilled,
        (state, { payload }) => {
          state.status = "succeeded";
          state.user = payload;
          state.logged_in = true;
        }
      )
      .addMatcher(api.endpoints.logOut.matchFulfilled, (state) => {
        state.status = "succeeded";
        state.user = undefined;
        state.logged_in = false;
      })
      .addMatcher(
        api.endpoints.createNewAccount.matchFulfilled,
        (state, { payload }) => {
          state.status = "succeeded";
          state.user = payload;
          state.logged_in = true;
        }
      )
      .addMatcher(
        api.endpoints.getCurrentAccount.matchFulfilled,
        (state, { payload }) => {
          state.status = "succeeded";
          state.user = payload;
          state.logged_in = payload.id ? true : false;
        }
      );
  },
});

export default authSlice.reducer;
