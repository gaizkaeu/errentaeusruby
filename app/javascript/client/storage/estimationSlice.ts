/**
 * Used to preserve the state of the current estimation saved on the
 * session (server side).
 *
 * Used to create the estimation (and save it server-side session)
 */
import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { AxiosError } from "axios";
import { createNewEstimation, myEstimation } from "./apiService";
import { EstimationState, Estimation, EstimationData } from "./types";

// Define the initial state using that type
const initialState = {
  status: "succeeded",
  error: undefined,
  estimation_props: undefined,
  estimation: undefined,
} as EstimationState;

const estimationSlice = createSlice({
  name: "estimations",
  initialState: initialState,
  reducers: {
    resetEstimation(state) {
      state.estimation = undefined;
      state.estimation_props = undefined;
    },
  },
  extraReducers(builder) {
    builder
      .addCase(calculateEstimation.pending, (state, action) => {
        state.status = "loading";
      })
      .addCase(calculateEstimation.fulfilled, (state, action) => {
        state.status = "succeeded";

        const [res, req] = action.payload;
        state.estimation = res;
        state.estimation_props = req;
      })
      .addCase(calculateEstimation.rejected, (state, action) => {
        state.status = "failed";
        if (action.payload) {
          state.error = action.payload.errorMessage;
        } else {
          state.error = action.error.message;
        }
      })
      .addCase(rescueMyEstimation.pending, (state, action) => {
        state.status = "loading";
      })
      .addCase(rescueMyEstimation.fulfilled, (state, action) => {
        state.status = "succeeded";
        if (action.payload) {
          state.estimation = action.payload;
        }
      })
      .addCase(rescueMyEstimation.rejected, (state, action) => {
        state.status = "failed";
        state.error = action.error.message;
      });
  },
});

export const calculateEstimation = createAsyncThunk<
  [Estimation, EstimationData],
  EstimationData,
  { rejectValue: Record<string, string> }
>("estimations/calculateEstimation", async (data, { rejectWithValue }) => {
  try {
    const response = await createNewEstimation(data);
    return [response, data];
  } catch (err) {
    const error = err as AxiosError<Record<string, string>>;
    if (!error.response) {
      throw err;
    }
    return rejectWithValue(error.response.data);
  }
});

export const rescueMyEstimation = createAsyncThunk<Estimation>(
  "estimations/rescueMyEstimation",
  async (data) => {
    const response = await myEstimation();
    return response;
  }
);

export default estimationSlice.reducer;
export const { resetEstimation } = estimationSlice.actions;
