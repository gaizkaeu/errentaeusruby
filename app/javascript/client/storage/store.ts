import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/dist/query";
import authSlice from "./authSlice";
import calculatorSlice from "./calculatorSlice";
import estimationSlice from "./estimationSlice";
import { api } from "./api";

export const store = configureStore({
  reducer: {
    estimations: estimationSlice,
    authentication: authSlice,
    calculator: calculatorSlice,
    [api.reducerPath]: api.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(api.middleware),
});

setupListeners(store.dispatch);
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
