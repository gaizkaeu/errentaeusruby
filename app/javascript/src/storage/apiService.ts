import axios from "axios";
import { CurrentUser, SessionCreationData, UserRegistrationData } from "./authSlice";
import { Estimation, EstimationData } from "./estimationSlice";
import { TaxIncome, TaxIncomeData } from "./taxIncomeSlice";

const API_BASE = "http://192.168.1.129:3000/"

const checkLoggedIn = async () => {
    const response = await axios.get<{user: CurrentUser, logged_in: boolean}>(API_BASE + 'api/v1/logged_in')
    return response.data;
}

const createNewUser = async (data: UserRegistrationData) => {
    const response = await axios.post<CurrentUser>(API_BASE + 'api/v1/users', {api_v1_user: data})
    return response.data;
}

const createNewSession = async (data: SessionCreationData) => {
    const response = await axios.post<CurrentUser>(API_BASE + 'api/v1/users/sign_in', {api_v1_user: data})
    return response.data;
}

const closeSession = async () => {
   const response = await axios.delete(API_BASE + '/api/v1/users/sign_out') 
   return response.status;
}

const createNewEstimation = async (data: EstimationData) => {
    const response = await axios.post<Estimation>(API_BASE + 'api/v1/estimations/estimate', data)
    return response.data;
}

const myEstimation = async () => {
    const response = await axios.get<Estimation>(API_BASE + 'api/v1/estimations/my_estimation')
    return response.data;
}

const createNewTaxIncome = async (data: TaxIncomeData) => {
    const response = await axios.post<TaxIncome>(API_BASE + 'api/v1/tax_incomes/', data)
    return response.data;
}

const listIncomeTaxes = async () => {
    const response = await axios.get<TaxIncome[]>(API_BASE + 'api/v1/tax_incomes/')
    return response.data;
}


export {createNewEstimation, checkLoggedIn, createNewUser, createNewSession, closeSession, myEstimation, createNewTaxIncome, listIncomeTaxes};