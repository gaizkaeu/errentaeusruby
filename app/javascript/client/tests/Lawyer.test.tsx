import renderer from "react-test-renderer";
import LawyerAvatar from "../components/Lawyer/Lawyer";
import { IUser } from "../storage/types";

it("Lawyer::Avatar: show correct data", () => {
  const user: IUser = {
    id: "0",
    email: "gaizkaurdangarin@gmail.com",
    name: "Gaizka",
    surname: "Urdangarin",
    account_type: "user",
  };
  const component = renderer.create(<LawyerAvatar lawyer={user} size="sm" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
