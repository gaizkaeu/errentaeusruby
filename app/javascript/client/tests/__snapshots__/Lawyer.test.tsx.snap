// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows correct data 1`] = `
<div
  className="nextui-c-ijHtNE nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-ijHtNE-iPJLV-css"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <span
    className="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-bHlgBq-size-sm nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar nextui-user-avatar"
    data-state="ready"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <span
      className="nextui-avatar-bg"
    />
    <span
      className="nextui-avatar-text"
    >
      Gai
    </span>
  </span>
  <div
    className="nextui-c-kNKVvs nextui-user-info"
  >
    <span
      className="nextui-c-hEcroZ nextui-user-name"
    >
      Gaizka
    </span>
    <span
      className="nextui-c-eCZQvV nextui-user-desc"
    />
  </div>
</div>
`;
